pydantic/_internal/_generate_schema.py

```diff
@@ -296,6 +296,15 @@ def push(self, for_type: type[Any]):
             self._types_namespace_stack.pop()
 
 
+def _get_first_non_null(a: Any, b: Any) -> Any:
```

Comment:
```suggestion
def _get_first_not_none(a: Any, b: Any) -> Any:
```
not a big deal.

Reply:
For what it's worth, I don't think it would be crazy to just copy the `if <a> is not None else` everywhere, but I'm okay with this too.